### ISSUE TITLE
Interface for melting data.frame?

### ISSUE THREAD

**Issue by benwhalley** [Rdatatable/data.table#4024]
I'm aware of the other issues that have been raised on this, but am trying to understand the rationale for removing the ability to melt a data.frame _entirely_, rather than simply switching away from the reshape dependency.

I have previously (perhaps wrongly) taught students to use data.table::melt on data frames. I understand the motivation to move away from reshape, but rather than breaking old code, would it be possible to coerce dataframes to data.table and use the new routines, obviously issuing a warning to the user as it happens?

**Comment by MichaelChirico** [comment]
Thanks for raising. This one is a lot easier to do (than adding `melt.list`, `melt.matrix`, etc methods), and I think we should support it.

**Comment by jangorecki** [comment]
Agree, providing data.frame method to dispatch seems to be best way
